### PR TITLE
Avoid unnecessary unboxing of Boolean

### PR DESCRIPTION
**Pull request by prasanth08**
Changed Boolean.valueOf() to Boolean.parseBoolean() - Boolean.valueOf() returns Boolean while the attributes in Options Class is boolean, Boolean.parseBoolean() returns boolean.
Changed Collections.sort() to List.sort() - Collections.sort() internally calls the List.sort().
Fix Java - doc

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/MappingProcessor.java [hunk 1]
```diff
@@ -63,7 +62,7 @@
  * <ul>
  * <li>retrieve mapping methods</li>
  * <li>create the {@code Mapper} model</li>
- * <li>perform enrichments and modifications (e.g. add annotations for dependency injection)</li>
+ * <li>perform enrichment's and modifications (e.g. add annotations for dependency injection)</li>
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/MappingProcessor.java:65]: Why do you think that we need to change this? Right now it is a plural noun of enrichment. With the change it becomes something else.

**Review by filiphr — CHANGES_REQUESTED** [review]
Your changes are OK @prasanth08, only one small comment about one of the Javadocs. 

Apart from that, can you please rebase this on top of main. I see that there are some conflicts

File: processor/src/main/java/org/mapstruct/ap/MappingProcessor.java (hunk 1 above)
Comment (prasanth08) [reply processor/src/main/java/org/mapstruct/ap/MappingProcessor.java]: I agree @filiphr  .Looks like the apostrophe came in from intellij and sonar plugin just before pushing the change.

**Comment by filiphr** [comment]
Thanks @prasanth08

**Pull request merged**